### PR TITLE
Start building with lts/dubnium (v10) and update supported node version to lts/carbon (v8)

### PR DESCRIPTION
**Pull request by taymoork2**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -21,5 +21,8 @@
     "node_modules/**": true,
     "packages/node_modules/**": false
   },
-  "eslint.autoFixOnSave": true
+  "eslint.autoFixOnSave": true,
```
Comment (adamweeks) [inline .vscode/settings.json]: Is this change required for the node upgrade?

Comment (taymoork2) [reply .vscode/settings.json]: Nope, vscode did that on it's own
Although I'm not opposed to adding it in as its own commit in the PR since it will probably show up in everyone elses PRs

Comment (adamweeks) [reply .vscode/settings.json]: Sounds good, let's put it in a commit.
